utils/src/parallel_restore.c

```diff
@@ -1832,13 +1832,53 @@ spr_err_t scoutfs_parallel_restore_write_buf(struct scoutfs_parallel_restore_wri
 	return count > 0 ? 0 : err;
 }
 
+/*
+ * Here we take in a dev's fd an read its quorum blocks to see if the dev has
+ * been mounted before
+ */
+static spr_err_t scoutfs_check_if_previous_mount(int fd)
+{
+	struct scoutfs_quorum_block *blk = NULL;
+	struct scoutfs_quorum_block_event *ev;
+	u64 blkno;
+	int i, j;
+	spr_err_t err;
+
+	for (i = 0; i <  SCOUTFS_QUORUM_MAX_SLOTS; i++) {
+		blkno = SCOUTFS_QUORUM_BLKNO + i;
+		free(blk);
```

Comment:
Generally you shouldn't do this and the NULL assignment right before the allocation. Since read_block allocates, you want to start with blk = NULL (and you properly initialize it above), but because you free(blk) before read_block() is called, blk will only be freed if there are more iterations in the loop. On the last iteration, blk is never freed and you leak the memory.

In other words, just do read_block() ... free() every loop in that order, and make sure to account for error cases.